memory-viz/src/memory_model.ts

```diff
@@ -614,6 +613,31 @@ export class MemoryModel {
                 this.getTextLength(idv + 5, style.text_value)
             );
 
+            box_width = Math.max(
+                box_width,
+                this.obj_x_padding * 2 +
+                    key_box +
+                    value_box +
+                    2 * this.font_size
+            );
+            box_height += 1.5 * this.item_min_height;
+        }
+
+        this.drawRect(x, y, box_width, box_height, style.box_container);
+
+        const SIZE: Rect = { x, y, width: box_width, height: box_height };
+
+        let curr_y = y + this.prop_min_height + this.item_min_height / 2;
```

Comment:
You can move this to underneath the comment on Line 632; it can be grouped with the loop since it's an accumulator.